docs/source/sparse.rst

```diff
@@ -1,146 +1,382 @@
+backward with respect to sparse matrix argument. All Torch operations,
+except :func:`torch.smm`, support backward with respect to strided
+matrix arguments.
 
 .. note::
 
-    Our sparse tensor format permits *uncoalesced* sparse tensors, where
-    there may be duplicate coordinates in the indices; in this case,
-    the interpretation is that the value at that index is the sum of all
-    duplicate value entries. Uncoalesced tensors permit us to implement
-    certain operators more efficiently.
-
-    For the most part, you shouldn't have to care whether or not a
-    sparse tensor is coalesced or not, as most operations will work
-    identically given a coalesced or uncoalesced sparse tensor.
-    However, there are two cases in which you may need to care.
-
-    First, if you repeatedly perform an operation that can produce
-    duplicate entries (e.g., :func:`torch.sparse.FloatTensor.add`), you
-    should occasionally coalesce your sparse tensors to prevent
-    them from growing too large.
-
-    Second, some operators will produce different values depending on
-    whether or not they are coalesced or not (e.g.,
-    :func:`torch.sparse.FloatTensor._values` and
-    :func:`torch.sparse.FloatTensor._indices`, as well as
-    :func:`torch.Tensor.sparse_mask`).  These operators are
-    prefixed by an underscore to indicate that they reveal internal
-    implementation details and should be used with care, since code
-    that works with coalesced sparse tensors may not work with
-    uncoalesced sparse tensors; generally speaking, it is safest
-    to explicitly coalesce before working with these operators.
-
-    For example, suppose that we wanted to implement an operator
-    by operating directly on :func:`torch.sparse.FloatTensor._values`.
-    Multiplication by a scalar can be implemented in the obvious way,
-    as multiplication distributes over addition; however, square root
-    cannot be implemented directly, since ``sqrt(a + b) != sqrt(a) +
-    sqrt(b)`` (which is what would be computed if you were given an
-    uncoalesced tensor.)
-
-.. class:: FloatTensor()
-
-    .. method:: add
-    .. method:: add_
-    .. method:: clone
-    .. method:: dim
-    .. method:: div
-    .. method:: div_
-    .. method:: get_device
-    .. method:: hspmm
-    .. method:: mm
-    .. method:: mul
-    .. method:: mul_
-    .. method:: narrow_copy
-    .. method:: resizeAs_
-    .. method:: size
-    .. method:: spadd
-    .. method:: spmm
-    .. method:: sspaddmm
-    .. method:: sspmm
-    .. method:: sub
-    .. method:: sub_
-    .. method:: t_
-    .. method:: to_dense
-    .. method:: transpose
-    .. method:: transpose_
-    .. method:: zero_
-    .. method:: coalesce
-    .. method:: is_coalesced
-    .. method:: _indices
-    .. method:: _values
-    .. method:: _nnz
-
-Functions
-----------------------------------
+   :func:`torch.spmm` and :func:`torch.dsmm` are aliases to :func:`torch.mm`.
```

Comment:
Per comment below, are these aliases useful?

Reply:
Not to my knowledge.

Reply:
Resolving this discussion via https://github.com/pytorch/pytorch/issues/46164 and removing references to spmm/dsmm from this document.